Minio/DataModel/MinioClientBuilder.cs

```diff
@@ -0,0 +1,188 @@
+/*
+ * MinIO .NET Library for Amazon S3 Compatible Cloud Storage,
+ * (C) 2020 MinIO, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using System.Net;
+using Minio.Exceptions;
+
+namespace Minio
+{
+    public interface IMinioClient
+    {
+        MinioClient Build();
+        MinioClient WithCredentials(string accessKey, string secretKey);
+        MinioClient WithRegion(string region);
+        MinioClient WithEndpoint(Uri url);
+        MinioClient WithEndpoint(string endpoint, int port);
+        MinioClient WithEndpoint(string endpoint);
+        MinioClient WithSessionToken(string sessiontoken);
+    }
+
+    public partial class MinioClient : IMinioClient
+    {
+        internal bool IsAwsHost { get; private set; }
+        internal bool IsAwsChinaHost { get; private set; }
+        internal bool IsAcceleratedHost { get; private set; }
+        internal bool IsDualStackHost { get; private set; }
+        internal bool UseVirtualStyle { get; private set; }
+        internal string RegionInUrl { get; private set; }
```

Comment:
RegionInUrl - why are these defined here? Most of the vars defined above are not being used - and can be removed.

Reply:
Removed.